_posts/2022-07-27-osbuild-operator-for-os-lifecycle-in-flotta.markdown

```diff
@@ -0,0 +1,204 @@
+oc new-app --env-file config/creating_env/psql.env postgresql:13-el8 -n osbuild
+```
+
+Edit the file `config/creating_env/postgress_secret.yaml` with the same Password
+
+Create new secret (please enter a real encoded password) 
+```shell
+oc create -f config/creating_env/postgress_secret.yaml
+```
+
+### External Worker VM using CNV ###
+Create an SSH key-pair
+```shell
+ssh-keygen -t rsa -b 4096 -C cloud-user@external-builder -f ~/.ssh/external-builder
+```
+
+Create symlinks to the files to facilitate the next step
+```shell
+ln -s ~/.ssh/external-builder.pub config/creating_env/ssh-publickey
+ln -s ~/.ssh/external-builder config/creating_env/ssh-privatekey
+```
+
+Generate the secret
+```shell
+oc create secret generic external-builder-ssh-pair --from-file=config/creating_env/ssh-privatekey --from-file=config/creating_env/ssh-publickey -n osbuild
+```
+
+Deploy the VM
+```shell
+oc apply -n osbuild -f config/creating_env/external-worker-vm.yaml
+```
+
+Get VM Address
+```shell
+oc get vmi external-builder -o jsonpath={.status.interfaces[0].ipAddress}
+```
+
+
+## Create OSBuildEnvConfig singleton CR ##
+Apply OSBuildEnvConfig
+```shell
+export CLUSTER_DOMAIN='mycluster.example.com'
+export EXTERNAL_WORKER_IP=`oc get vmi external-builder -o jsonpath={.status.interfaces[0].ipAddress}`
+cat config/samples/osbuilder_v1alpha1_osbuildenvconfig.yaml | envsubst | oc apply -f -
+```
+
+## Create edge-container image ##
+Apply osbuildConfig that contains all the relevant configuration for running osbuild-composer job.
+You can use that sample:
+```shell
+oc create -f config/samples/osbuilder_v1alpha1_osbuildconfig.yaml -n osbuild
+```
+
+After applying osbuildConfig CR you should see a new osbuild CR that was triggered automatically
+```shell
+oc describe osbuild osbuildconfig-sample-1 -n osbuild 
+```
+
+As part of that osbuild CR you can found the osbuild-composer job status, the job composer ID, and after finish running you also can found the containerUrl in the S3 bucket
+```shell
+  status:
+    conditions:
+    - message: Edge-container job is still running
+      status: ""
+      type: startedContainerBuild
+    - message: Edge-container job was finished successfully
+      status: ""
+      type: containerBuildDone
+    containerComposeId: 80125a8c-6e4a-42fe-8c06-8f90398fe5b1
+    containerUrl: https://minio-s3-osbuild.apps.my-cluster.example.com/osbuild-images/composer-api-2ca271b8-1a9e-4bae-bcec-ab151c2bdc5b-container.tar?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minioadmin%2F20220720%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220720T095954Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=ac57b8499450dd4b62b80936277db89e7e850d5c3efab9e39bb9a493c064921f
+```
+
+## Run VM using that edge-container ##
+
+### Upload the tar file to container image registry ###
+Download the tar file to your local environment 
+```shell
+curl -k <containerUrl> -o /tmp/local.tar
+```
+
+Upload it to pulp container registry using skopeo
```

Comment:
The Container Registry does not have to by pulp, it can be any registry (docker, quay, etc.). So please rephrase "Upload the image to your Container Registry using skopeo"